### PR TITLE
refactor: migrate search

### PR DESCRIPTION
**Pull request by kyubisation**

### CODE REVIEW

File: src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+export class SearchCustomIconAutocompleteStaticOptionsExample implements OnInit {
+  searchControl = new FormControl('');
+
+  staticOptions: string[] = ['static option one', 'static option two'];
+  options = new Subject<string[]>();
+
+  ngOnInit() {
+    this.searchControl.valueChanges
+      .pipe(debounceTime(500))
+      .pipe(distinctUntilChanged())
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts:23]: takeUntil destroyed

File: src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  constructor(private _http: HttpClient) {}
+
+  ngOnInit() {
+    this.searchControl.valueChanges.subscribe((newValue) => {
+      this.filteredCities = this.cities.filter(
+        (option) => option.toLocaleUpperCase().indexOf(newValue.toLocaleUpperCase()) > -1
+      );
+    });
+
+    this.searchSubject.pipe(distinctUntilChanged()).subscribe((searchTerm) => {
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts:43]: takeUntil destroyed

File: src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    'St. Gallen',
+    'Lugano',
+    'Thun',
+  ];
+  filteredCities = this.cities.slice(0);
+
+  constructor(private _http: HttpClient) {}
+
+  ngOnInit() {
+    this.searchControl.valueChanges.subscribe((newValue) => {
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts:37]: takeUntil destroyed

File: src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ */
+@Component({
+  selector: 'sbb-search-autocomplete-example',
+  templateUrl: './search-autocomplete-example.html',
+})
+export class SearchAutocompleteExample {
+  searchControl = new FormControl('');
+  searchValues: string[] = [];
+
+  filteredOptions = options.slice(0);
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts:16]: there are no "filtered" options here, so using options directly should work

File: src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  handleSearch(value: string) {
+    this.searchValues = [value, ...this.searchValues].slice(0, 5);
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts:19]: I think slicing is an overkill for the showcase examples and can only confuse consumers?

File: src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  templateUrl: './search-header-autocomplete-example.html',
+})
+export class SearchHeaderAutocompleteExample {
+  searchValues: string[] = [];
+  searchControl = new FormControl('');
+
+  filteredOptions = options.slice(0);
+
+  handleSearch(value: string) {
+    this.searchValues = [value, ...this.searchValues].slice(0, 5);
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts:19]: I think slicing is an overkill for the showcase examples and can only confuse consumers?

File: src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+ */
+@Component({
+  selector: 'sbb-search-header-autocomplete-example',
+  templateUrl: './search-header-autocomplete-example.html',
+})
+export class SearchHeaderAutocompleteExample {
+  searchValues: string[] = [];
+  searchControl = new FormControl('');
+
+  filteredOptions = options.slice(0);
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts:16]: there are no "filtered" options here, so using options directly should work

File: src/components-examples/angular/search/search-simple-header-mode/search-simple-header-mode-example.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+@Component({
+  selector: 'sbb-search-simple-header-mode-example',
+  templateUrl: './search-simple-header-mode-example.html',
+})
+export class SearchSimpleHeaderModeExample {
+  searchValues: string[] = [];
+  searchControl = new FormControl('');
+
+  handleSearch(value: string) {
+    this.searchValues = [value, ...this.searchValues].slice(0, 5);
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-simple-header-mode/search-simple-header-mode-example.ts:17]: I think slicing is an overkill for the showcase examples and can only confuse consumers?

File: src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts [hunk 9]
```diff
@@ -0,0 +1,49 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged } from 'rxjs/operators';
+
+/**
+ * @title Custom Icon
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts:7]:
```suggestion
 * @title Custom Icon And Autocomplete With Static Options
```

File: src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+ * @order 20
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-custom-icon-autocomplete-static-options/search-custom-icon-autocomplete-static-options-example.ts:8]:
```suggestion
 * @order 30
```

File: src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
+import { Component } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+/**
+ * @title Search Header Mode With Autocomplete
+ * @order 30
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts:6]:
```suggestion
 * @order 50
```

File: src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts [hunk 12]
```diff
@@ -0,0 +1,69 @@
+import { HttpClient } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Subject } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+/**
+ * @title Search Historic Railway Pictures
+ * @order 50
```
Comment (jeripeierSBB) [inline src/components-examples/angular/search/search-historic-railway-pictures/search-historic-railway-pictures-example.ts:9]:
```suggestion
 * @order 60
```

File: src/angular/search/search-animation.ts [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+} = {
+  /**
+   * This animation transforms the select's overlay panel on and off the page.
+   *
+   * When the panel is attached to the DOM, it fades in and when the panel is removed, it fades out.
+   */
+  growShrink: trigger('growShrink', [
+    state('void', style({ opacity: 0, width: '0px' })),
+    state('open', style({ opacity: 1, width: '*' })),
+    transition('void => *', animate('120ms cubic-bezier(0, 0, 0.2, 1)')),
```
Comment (jeripeierSBB) [inline src/angular/search/search-animation.ts:26]:
```suggestion
    transition('void => *', animate('300ms ease')),
```

File: src/angular/search/search-animation.ts [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+    transition('* => void', animate('100ms 25ms linear')),
```
Comment (jeripeierSBB) [inline src/angular/search/search-animation.ts:27]:
```suggestion
    transition('* => void', animate('300ms 25ms linear')),
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts (hunk 5 above)
Comment (kyubisation) [reply src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts:19]: I don't want the list to grow unlimited. Should I maybe add a comment to explain why this is done?

File: src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts (hunk 6 above)
Comment (kyubisation) [reply src/components-examples/angular/search/search-header-autocomplete/search-header-autocomplete-example.ts:19]: I don't want the list to grow unlimited. Should I maybe add a comment to explain why this is done?

File: src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts (hunk 5 above)
Comment (jeripeierSBB) [reply src/components-examples/angular/search/search-autocomplete/search-autocomplete-example.ts:19]: Also okay for me

**Review by jeripeierSBB — APPROVED** [review]
LGTM (G stays for Great :-) )

**Pull request merged**